README.md

```diff
@@ -355,6 +356,75 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+#### Why use Interface instead of Type for typing props?
+
+`interface` are a little more powerful than the corresponding `type` declaration.
+
+One of the reasons to use `interface` is when you need to compose two or more types, you have the
+option of extending those types with an interface, or intersecting them in a type alias. TS team
+says:
+
+> Interfaces create a single flat object type that detects property conflicts, which are usually
+> important to resolve! Intersections on the other hand just recursively merge properties, and in
+> some cases produce never. Interfaces also display consistently better, whereas type aliases to
+> intersections can't be displayed in part of other intersections. Type relationships between
+> interfaces are also cached, as opposed to intersection types as a whole. A final noteworthy
+> difference is that when checking against a target intersection type, every constituent is checked
+> before checking against the "effective"/"flattened" type. For this reason, extending types with
+> interfaces/extends is suggested over creating intersection types.
+
+This is preferred:
+
+```ts
+interface Foo extends Bar, Baz {
+  someProp: string;
+}
+```
+
+Instead of this:
+
+```ts
+type Foo = Bar &
```

Comment:
Maybe we should mention when it is good to use a `type`? Eg. when you want to use OR (`|`):
```
type ButtonProps = {
  label?: string;
  onClick?: (event: MouseEvent) => void;
} & (
  | {
      href: HTMLAnchorElement['href'];
      target?: HTMLAnchorElement['target'];
      type?: HTMLAnchorElement['type'];
    }
  | {
      href?: never;
      target?: never;
      type?: HTMLButtonElement['type'];
    }
);
```

Reply:
@ThijsTyZ make sense and I agree. But it would not be good if we move those 2 extra types properties for the attributes into two separate interfaces and then use (`|`)?

```ts
const Button: ButtonProps & (ButtonAttributeProps1 | ButtonAttributeProps2) = {
  href: "test.com"
};
```
Link: https://codesandbox.io/s/recursing-kalam-l0h9n?file=/src/index.ts:341-442

PS: I don't have context of why you used that approach, I would like to learn haha. If we can't achieve the same goal you used on your example using Interface, then it would good to mention when it is good to use `type`

Reply:
This code example is coming from a real project. It is a CTA component that could render as an `<a>` tag or as a `<button>` tag. If you provide an `href` prop, it should render as a `<a>` and then you have to option to specify a `target`. If you do not specify an `href` prop it would render as a `<button>` tag. Both tags support a `type` prop, but with different values.

Reply:
This is not related to the PR